### PR TITLE
Remove obsolete parameters from analyze rest spec

### PR DESCRIPTION
**Pull request by romseygeek**
This commit also fixes a typo in the analyze high-level client
documentation.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.analyze.json [hunk 1]
```diff
@@ -15,16 +15,6 @@
         "index": {
```
Comment (romseygeek) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.analyze.json]: I've left the "index" parameter in here, despite it being a path param, as IIUC then path params and URL params get collapsed into a single params object in the rest handler.

Comment (javanna) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.analyze.json]: this is correct, index should be left as-is.

**Review by javanna — APPROVED** [review]
LGTM thanks @romseygeek !

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**